ace-isearch.el

```diff
@@ -218,16 +218,15 @@ of `isearch-string' is longer than or equal to `ace-isearch-input-length'."
            2)))
     (cond (;; using avy/ace-jump since L=1 or L=2 reached (depending on `ace-isearch-jump-based-on-one-char')
            (and (= (length isearch-string) ace-isearch-input-min-length)
-                (and (not isearch-regexp)
-                     (or (not (ace-isearch--isearch-regexp-function))
+                (and (or (not (ace-isearch--isearch-regexp-function))
```

Comment:
Good fix. 
By introducing the variable below, 
```elisp
(defcustom ace-isearch-on-evil-mode nil
  "If non nil, ace-isearch-mode can be used on Evil mode."
  :type 'boolean
  :group 'ace-isearch)
```
the code can be fixed as follows:

```elisp
 (and (if ace-isearch-on-evil-mode
         t
       (not isearch-regexp))
     (or (not (ace-isearch--isearch-regexp-function))
         (not (eq search-default-mode nil))))
```